pydantic/schema.py

```diff
@@ -379,15 +381,15 @@ def get_flat_models_from_field(field: Field) -> Set[Type['main.BaseModel']]:
     flat_models: Set[Type['main.BaseModel']] = set()
     if field.sub_fields:
         flat_models |= get_flat_models_from_fields(field.sub_fields)
-    elif lenient_issubclass(field.type_, main.BaseModel):
+    elif lenient_issubclass(field.type_, pydantic.main.BaseModel):
```

Comment:
maybe you could add

```py
BaseModel = pydantic.main.BaseModel
```
To avoid having to access `pydantic.main.BaseModel` so many times?